src/KernelAbstractions.jl

```diff
@@ -457,12 +457,22 @@ include("macros.jl")
 # Backends/Interface
 ###
 
-function Scratchpad(ctx, ::Type{T}, ::Val{Dims}) where {T, Dims}
-    throw(MethodError(Scratchpad, (T, Val(Dims))))
+@inline function SharedMemory(::Type{T}, ::Val{Dims}, ::Val) where {T, Dims}
```

Comment:
Can we leave the implementation back to CPU.jl and here just define:

```
function Scratchpad end
```

Reply:
I changed it and also added `function SharedMemory end`.